tests/test_data.py

```diff
@@ -145,17 +153,125 @@ async def test_different_conf_thresholds():
     assert not disputable
+
+
+@pytest.mark.asyncio
+async def test_feed_filter(caplog, log):
+    """test that, if the user wants to monitor only one feed, all other feeds are ignored and skipped"""
+
+    caplog.set_level(logging.INFO)
+
+    cfg = TelliotConfig()
+    cfg.main.chain_id = 1
+
+    for endpoint in cfg.endpoints.find(chain_id=1):
+        cfg.endpoints.endpoints.remove(endpoint)
+
+    endpoint = RPCEndpoint(
+        1, "Mainnet", "Infura", "https://mainnet.infura.io/v3/db7ce830b1224efe93ae3240f7aaa764", "etherscan.io"
+    )
+    cfg.endpoints.endpoints.append(endpoint)
+
+    # we are monitoring a different feed
+    feed = btc_usd_median_feed
+
+    confidence_threshold = 0.50
+
+    # try to parse it
+    res = await parse_new_report_event(cfg, confidence_threshold, feed=feed, log=log)
+
+    # parser should return None
+    assert not res
+    assert "skipping undesired NewReport event" in caplog.text
+
+    cfg.endpoints.endpoints.remove(endpoint)
+
+
+@pytest.mark.asyncio
+async def test_parse_oracle_address_submission():
+
+    update_oracle_address_query_id = "0xcf0c5863be1cf3b948a9ff43290f931399765d051a60c3b23a4e098148b1f707"
+
+    log = AttributeDict(
+        {
+            "address": "0xD9157453E2668B2fc45b7A803D3FEF3642430cC0",
+            "topics": [
+                HexBytes("0x48e9e2c732ba278de6ac88a3a57a5c5ba13d3d8370e709b3b98333a57876ca95"),
+                HexBytes(update_oracle_address_query_id),
+                HexBytes("0x0000000000000000000000000000000000000000000000000000000063caf0f7"),
+                HexBytes("0x00000000000000000000000050a86759d495ecfa7c301071d6b0bdd4bd664ab0"),
+            ],
+            "data": "0x0000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000d9157453e2668b2fc45b7a803d3fef3642430cc000000000000000000000000000000000000000000000000000000000000000e000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000001354656c6c6f724f7261636c654164647265737300000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000",  # noqa: E501
+            "blockNumber": 16450223,
+            "transactionHash": HexBytes("0xa5cac44128bbe2c195ed9dbc2412e8fb2e97b960a9aeb49c2ac111d35603579a"),
+            "transactionIndex": 16,
+            "blockHash": HexBytes("0xcf720f24b1793eb4d58c6137535e6512b7e62794d1ef4db61eff9dc7c67ef03b"),
+            "logIndex": 63,
+            "removed": False,
+        }
+    )
+
+    class TellorOracleAddress(AbiQuery):
```

Comment:
Can import this class from `telliot-feeds` instead?

Reply:
Sure, if it is included in a release...if not, i just mocked the class as seen here